packages/zowe-explorer/src/trees/dataset/DatasetFSProvider.ts

```diff
@@ -380,7 +389,28 @@ export class DatasetFSProvider extends BaseProvider implements vscode.FileSystem
      * @returns The data set's contents as an array of bytes
      */
     public async readFile(uri: vscode.Uri): Promise<Uint8Array> {
-        const file = this._lookupAsFile(uri);
+        let ds: DsEntry | DirEntry;
```

Comment:
curious if it would be possible to combine some of the try/catch logic between the DS and USS file providers on their readFile functions 😋 

it feels somewhat similar 😋

Reply:
I agree, I think some of this logic can be consolidated into "abstract" filesystem functions and reused elsewhere. While I'd like to look into that, it'll probably expand the scope of the PR, so I can create an issue to track that 🙂

Reply:
Filed as #3099